pylace/src/utils.rs

```diff
@@ -559,44 +559,46 @@ fn df_to_values(
     indexer: &Indexer,
     engine: &lace::Engine,
 ) -> PyResult<DataFrameComponents> {
-    let row_names = if df.hasattr("index")? {
-        let index = df.getattr("index")?;
-        srs_to_strings(index).ok()
-    } else {
-        df.call_method1("__getitem__", ("index",))
-            .and_then(srs_to_strings)
-            .ok()
-    };
-
     Python::with_gil(|py| {
-        let (columns, data) = {
+        let (columns, data, row_names) = {
             let columns = df.getattr("columns").unwrap();
             if columns.get_type().name().unwrap().contains("Index") {
                 // Is a Pandas dataframe
+                let index = df.getattr("index")?;
+                let row_names = srs_to_strings(index).ok();
+
                 let cols =
                     columns.call_method0("tolist").unwrap().to_object(py);
                 let kwargs = PyDict::new(py);
                 kwargs.set_item("orient", "records").unwrap();
                 let data = df.call_method("to_dict", (), Some(kwargs)).unwrap();
-                (cols, data)
+                (cols, data, row_names)
             } else {
                 // Is a Polars dataframe
                 let list = columns.downcast::<PyList>().unwrap();
-                let has_index = list
+                let index_col_name = list
                     .iter()
-                    .any(|s| s.extract::<&str>().unwrap() == "index");
+                    .map(|s| s.extract::<&str>().unwrap())
+                    .find(|s| is_index_col(s));
```

Comment:
Do we need to consider the possibility of multiple index columns in a Polars dataframe?